src/StudentsController.ts

```diff
@@ -165,7 +169,36 @@ export class StudentsController {
         await this.#studentsCollection.update(oldDoc, student.toJSON());
     }
 
+    public async pseudonymizeStudent(student: Student): Promise<Student> {
+        // Only keep id and status of student (also works for future student properties)
+        const oldDoc = await this.#studentsCollection.read(student.id.toString());
+        const pseudonymizedDoc = {
+            id: oldDoc.id,
+            status: "deleted"
+        };
+
+        await this.#studentsCollection.update(oldDoc, pseudonymizedDoc);
+
+        return Student.from(pseudonymizedDoc);
+    }
+
     public async deleteStudent(student: Student): Promise<void> {
+        if (student.correspondingRelationshipId) {
+            const relationship = await this.services.transportServices.relationships.getRelationship({ id: student.correspondingRelationshipId.toString() });
+
+            // TODO: What about
+            // RelationshipStatus.Pending
+            // RelationshipStatus.Rejected
+            // RelationshipStatus.Revoked
```

Comment:
You can decompose rejected and revoked relationships. Pending ones would need to be rejected/revoked before you can decompose them.